src/name_core.jl

```diff
@@ -71,6 +71,30 @@ Base.@pure function dim_noerror(dimnames::Tuple{Vararg{Symbol, N}}, name::Symbol
     return 0
 end
 
+"""
+    expand_dimnames(dimnames, name)
+
+For `dimnames` being a tuple of names (symbols) for the dimensions.
+and `name` being a name.
+This expands the `dimnames` if `name` is not in `dimnames`.
+e.g. `expand_dimnames((:a, :b), :c) == (:a, :b, :c)`
+If `name` is already in `dimnames` then `dimnames` is returned.
+"""
+function expand_dimnames(dimnames::Tuple, name::Symbol)
+    if name in dimnames
+        return dimnames
+    else
+        return (dimnames..., name)
+    end
+end
+
+function expand_dimnames(dimnames::Tuple, name::Union{Integer, Colon, Tuple{}})
+    return dimnames
+end
```

Comment:
Good catch. I wasn't sure whether `expand_dimnames` is the way to go, so implemented the simplest possible approach (following YAGNI). I agree the current approach is a bit unintuitive.

How should `expand_dimnames` behave when called with an integer argument?

I guess it could be expanded with a `:_` if the number is higher than the existing number of dimensions. I.e. something like
```
julia> NamedDims.expand_dimnames((:x, :y), 1)
(:x, :y)

julia> NamedDims.expand_dimnames((:x, :y), 3)
(:x, :y, :_)
```

Reply:
that sounds right to me, yes.

Reply:
Done, and checked that there is no allocation.